### PR TITLE
[Inductor] Fix bug in MutationOutput IR

### LINKED ISSUE #127402: [inductor][cpu]abnormal performance improvement and accuracy drop for huggingface suit static/dynamic quantization
### 🐛 Describe the bug

abnormal performance improvement and accuracy drop for huggingface suit static/dynamic quantization in 2024-05-26 nightly release

accuracy drop:
</head>
<body link=blue vlink=purple>

model_name | static_quant_new | dynamic_quant_new | static_quant_old | dynamic_quant_old | static_quant(new/old) | dynamic_quant(new/old)
-- | -- | -- | -- | -- | -- | --
text-classification+albert-base-v1 | 0.3162 | 0.3162 | 0.5613 | 0.576 | 0.56 | 0.55
text-classification+bert-base-chinese | 0.3162 | 0.3162 | 0.6838 | 0.6838 | 0.46 | 0.46
text-classification+bert-large-cased | 0.3162 | 0.3162 | 0.6176 | 0.5466 | 0.51 | 0.58
text-classification+prajjwal1-bert-tiny | 0.3162 | 0.3162 | 0.4951 | 0.5 | 0.64 | 0.63
text-classification+xlnet-base-cased | 0.3971 | 0.4289 | 0.6765 | 0.6838 | 0.59 | 0.63

</body>
</html>

performance improve:
</head>
<body link=blue vlink=purple>

model_name | static_quant_new | dynamic_quant_new | static_quant_old | dynamic_quant_old | static_quant(new/old) | dynamic_quant(new/old)
-- | -- | -- | -- | -- | -- | --
text-classification+albert-base-v1 | 195.61 | 148.97 | 48.05 | 25.73 | 4.07 | 5.79
text-classification+bert-base-chinese | 184.7 | 140.91 | 47.81 | 24.47 | 3.86 | 5.76
text-classification+bert-large-cased | 121.71 | 102.55 | 20.18 | 11.33 | 6.03 | 9.05
text-classification+prajjwal1-bert-tiny | 264.08 | 247.91 | 194.85 | 177.77 | 1.36 | 1.39
text-classification+roberta-base | 189.79 | 148.74 | 47.58 | 25.71 | 3.99 | 5.79
text-classification+xlnet-base-cased | 28.85 | 27.55 | 23.07 | 17.03 | 1.25 | 1.62

</body>
</html>

f8c4c268da67e9684f3287b7468f36a5a27c6a0b
text-classification+bert-base-chinese
image
text-classification+albert-base-v1
image

ee6cb6daa173896f8ea1876266a19775aaa4f610 
text-classification+bert-base-chinese
image
text-classification+albert-base-v1
image

### Versions

</table><p>SW info</p><table border="1" class="dataframe table">
  <thead>
    <tr style="text-align: right;">
      <th>SW</th>
      <th>Branch</th>
      <th>Target commit</th>
      <th>Refer commit</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Pytorch</td>
      <td>nightly</td>
      <td>549167f</td>
      <td>bca6d8b</td>
    </tr>
    <tr>
      <td>Torchbench</td>
      <td>chuanqiw/inductor_quant</td>
      <td>ee35d764</td>
      <td>ee35d764</td>
    </tr>
    <tr>
      <td>torchaudio</td>
      <td>nightly</td>
      <td>1980f8a</td>
      <td>1980f8a</td>
    </tr>
    <tr>
      <td>torchtext</td>
      <td>nightly</td>
      <td>b0ebddc</td>
      <td>b0ebddc</td>
    </tr>
    <tr>
      <td>torchvision</td>
      <td>nightly</td>
      <td>d23a6e1</td>
      <td>d23a6e1</td>
    </tr>
    <tr>
      <td>torchdata</td>
      <td>nightly</td>
      <td>11bb5b8</td>
      <td>11bb5b8</td>
    </tr>
    <tr>
      <td>dynamo_benchmarks</td>
      <td>nightly</td>
      <td>nightly</td>
      <td>nightly</td>
    </tr>
  </tbody>

</table>

Repro:
```
git clone -b test https://github.com/chuanqi129/transformers && cd transformers && \
    python setup.py bdist_wheel && pip install --force-reinstall dist/*.whl && cd ..
git clone -b test https://github.com/zxd1997066/accelerate.git && cd accelerate && \
    python setup.py bdist_wheel && pip install --no-deps --force-reinstall dist/*.whl && cd ..
pip install -r transformers/examples/pytorch/text-classification/requirements.txt
wget https://github.com/chuanqi129/inductor-tools/raw/xiangdong/accuracy/scripts/modelbench/numa_launcher.py
wget https://github.com/chuanqi129/inductor-tools/raw/xiangdong/accuracy/scripts/modelbench/hf_quant_test.sh
#change model in https://github.com/chuanqi129/inductor-tools/blob/xiangdong/accuracy/scripts/modelbench/hf_quant_test.sh#L88
#static quantization
bash hf_quant_test.sh key torch_compile_quant_static
#dynamic quantization
bash hf_quant_test.sh key torch_compile_quant
```

Suspected guilty commit: https://github.com/pytorch/pytorch/commit/ee6cb6daa173896f8ea1876266a19775aaa4f610
[text-classification+albert-base-v1-static-quant-accuracy-drop_guilty_commit.log](https://github.com/pytorch/pytorch/files/15486362/text-classification%2Balbert-base-v1-static-quant-accuracy-drop_guilty_commit.log)
cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @bdhirsh @anijain2305 @chauhang @WeizhuoZhang-intel @chuanqi129

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Fixes #127402

https://github.com/pytorch/pytorch/pull/127151 made some changes to the MutationOutput IR. In the constructor, an argument is missing when calling its parent's constructor.

The issue can be reproduced by the following script (run with `TORCHINDUCTOR_FREEZING=1`):
```python
import torch
from torch._export import capture_pre_autograd_graph
from torch.ao.quantization.quantize_pt2e import (
    convert_pt2e,
    prepare_pt2e,
)
from torch.ao.quantization.quantizer.x86_inductor_quantizer import X86InductorQuantizer
import torch.ao.quantization.quantizer.x86_inductor_quantizer as xiq

class M(torch.nn.Module):
    def __init__(
        self,
        add_fn,
    ):
        super().__init__()
        self.linear1 = torch.nn.Linear(4, 4)
        self.linear2 = torch.nn.Linear(4, 4)
        self.add_fn = add_fn
        self.relu = torch.nn.ReLU()
        self.linear3 = torch.nn.Linear(4, 4)
        self.linear4 = torch.nn.Linear(4, 4)
        self.add_fn2 = add_fn
        self.relu2 = torch.nn.ReLU()

    def forward(self, x):
        x1 = self.linear1(x)
        x2 = self.linear2(x)
        tmp = self.add_fn(x1, x2)
        return tmp
        # tmp1 = self.linear3(tmp)
        # tmp2 = self.linear4(tmp)
        # res = self.add_fn2(tmp1, tmp2)
        # return res

add_fn = lambda x, y: x.add_(y)
mod = M(add_fn)
x = torch.randn((4, 4), dtype=torch.float32, requires_grad=False).add(1)
inputs = (x,)

with torch.no_grad():
    export_model = capture_pre_autograd_graph(mod, inputs)
    quantizer = X86InductorQuantizer()
    quantizer.set_global(
        xiq.get_default_x86_inductor_quantization_config()
    )

    prepare_model = prepare_pt2e(export_model, quantizer)
    prepare_model(*inputs)
    convert_model = convert_pt2e(prepare_model)
    torch.ao.quantization.move_exported_model_to_eval(convert_model)
    print(convert_model)

    out_ref = convert_model(*inputs)
    compiled_model = torch.compile(convert_model)
    compiled_model(*inputs)
    out = compiled_model(*inputs)
    print(torch.allclose(out_ref, out))
    print(torch.abs(out_ref - out).max())
```

Without this fix, a linear op is missing in the generated code.
Generated code without the fix:
```python
def call(args):
    arg8_1, = args
    args.clear()
    assert_size_stride(arg8_1, (4, 4), (4, 1))
    buf0 = empty_strided_cpu((4, 4), (4, 1), torch.uint8)
    buf1 = empty_strided_cpu((4, ), (1, ), torch.float32)
    buf2 = empty_strided_cpu((4, ), (1, ), torch.float32)
    buf3 = empty_strided_cpu((4, ), (1, ), torch.int64)
    cpp_fused_quantize_per_tensor_0(arg8_1, buf0, buf1, buf2, buf3)
    del arg8_1
    buf4 = torch.ops.onednn.qlinear_pointwise.default(buf0, 0.012747906148433685, 82, _frozen_param9, buf2, buf3, buf1, 1.0, 0, torch.float32, 'none', [None], '')
    assert_size_stride(buf4, (4, 4), (4, 1))
    del buf0
    del buf1
    del buf2
    del buf3
    return (buf4, )
```
Generated code with the fix:
```python
def call(args):
    arg8_1, = args
    args.clear()
    assert_size_stride(arg8_1, (4, 4), (4, 1))
    buf0 = empty_strided_cpu((4, 4), (4, 1), torch.uint8)
    buf1 = empty_strided_cpu((4, ), (1, ), torch.float32)
    buf2 = empty_strided_cpu((4, ), (1, ), torch.float32)
    buf3 = empty_strided_cpu((4, ), (1, ), torch.int64)
    cpp_fused_quantize_per_tensor_0(arg8_1, buf0, buf1, buf2, buf3)
    del arg8_1
    buf4 = torch.ops.onednn.qlinear_pointwise.default(buf0, 0.016188886016607285, 46, _frozen_param9, buf2, buf3, buf1, 1.0, 0, torch.float32, 'none', [None], '')
    assert_size_stride(buf4, (4, 4), (4, 1))
    buf5 = buf2; del buf2  # reuse
    buf6 = buf1; del buf1  # reuse
    buf7 = buf3; del buf3  # reuse
    cpp_fused_1(buf5, buf6, buf7)
    buf8 = torch.ops.onednn.qlinear_pointwise.binary(buf0, 0.016188886016607285, 46, _frozen_param8, buf6, buf7, buf5, 1.0, 0, torch.float32, buf4, 1.0, 0, 'sum', 1.0, 'none', [None], '')
    del buf0
    del buf5
    del buf6
    del buf7
    return (buf4, )
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by leslie-fang-intel — APPROVED** [review]

**Review by peterbell10 — CHANGES_REQUESTED** [review]
Can you add a test?

**Comment by Chillee** [comment]
Hm... that change was actually deliberate. Is the breakage a correctness or performance issue?

**Review by Chillee — CHANGES_REQUESTED** [review]
Hm... that change was actually deliberate. Is the breakage a correctness or performance issue?

**Comment by leslie-fang-intel** [comment]
Hi @Chillee, I think it breaks the correctness.

**Comment by leslie-fang-intel** [comment]
One another Question is `ConvolutionBinaryInplace` doesn't have similar issue. It looks like a `QLinearPointwiseBinaryPT2E` specific issue. Could we further compare what's the difference between these 2 IRs?

**Comment by Xia-Weiwen** [comment]
@leslie-fang-intel  @Chillee  Thanks for the inputs. The issue will be fixed via a change to QLinearPointwiseBinaryPT2E IR: https://github.com/pytorch/pytorch/pull/127592. So, close this PR.
